src/sentry/api/endpoints/sentry_internal_app_token_details.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import absolute_import
+
+from rest_framework.response import Response
+from rest_framework import status
+from django.http import Http404
+
+from sentry.api.bases import (
+    SentryInternalAppTokenPermission, SentryAppBaseEndpoint,
+)
+from sentry.models import ApiToken
+from sentry.features.helpers import requires_feature
+from sentry.mediators.sentry_app_installation_tokens import Destroyer
+
+
+class SentryInternalAppTokenDetailsEndpoint(SentryAppBaseEndpoint):
+    permission_classes = (SentryInternalAppTokenPermission, )
+
+    def convert_args(self, request, sentry_app_slug, token, *args, **kwargs):
+        # get the sentry_app from the SentryInternalAppTokenDetailsEndpoint class
+        (args, kwargs) = super(SentryInternalAppTokenDetailsEndpoint,
+                               self).convert_args(request, sentry_app_slug, *args, **kwargs)
+
+        try:
+            kwargs['api_token'] = ApiToken.objects.get(token=token)
```

Comment:
Should this be scoped to either the organization or sentry application? Right now it looks like you'd be able to get at tokens from outside your current organization.

Reply:
@markstory At this point, that is correct. However, if you look into the implementation of the `delete` method, you'll see we check that the API token is matched to the Sentry app of that route. That will prevent being able to get tokens from outside your organization.